File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/ElectricityDeliveryUnitFirm.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ElectricityDeliveryUnitFirm : ISerialized
    {
        public ElectricityDeliveryUnitFirm(XmlNode xmlNode)
        {
            XmlNode applicableNode = xmlNode.SelectSingleNode("applicable");
            
            if (applicableNode != null)
            {
                if (applicableNode.Attributes["href"] != null || applicableNode.Attributes["id"] != null) 
                {
                    if (applicableNode.Attributes["id"] != null) 
                    {
                        applicableIDRef_ = applicableNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(applicableNode);
                        IDManager.SetID(applicableIDRef_, ob);
                    }
                    else if (applicableNode.Attributes["href"] != null)
                    {
                        applicableIDRef_ = applicableNode.Attributes["href"].Value;
                    }
                    else
                    {
                        applicable_ = new XsdTypeBoolean(applicableNode);
                    }
                }
                else
                {
                    applicable_ = new XsdTypeBoolean(applicableNode);
                }
            }
            
        
            XmlNode generationAssetNode = xmlNode.SelectSingleNode("generationAsset");
            
            if (generationAssetNode != null)
            {
                if (generationAssetNode.Attributes["href"] != null || generationAssetNode.Attributes["id"] != null) 
                {
                    if (generationAssetNode.Attributes["id"] != null) 
                    {
                        generationAssetIDRef_ = generationAssetNode.Attributes["id"].Value;
                        CommodityDeliveryPoint ob = new CommodityDeliveryPoint(generationAssetNode);
                        IDManager.SetID(generationAssetIDRef_, ob);
                    }
                    else if (generationAssetNode.Attributes["href"] != null)
                    {
                        generationAssetIDRef_ = generationAssetNode.Attributes["href"].Value;
                    }
                    else
                    {
                        generationAsset_ = new CommodityDeliveryPoint(generationAssetNode);
                    }
                }
                else
                {
                    generationAsset_ = new CommodityDeliveryPoint(generationAssetNode);
                }
            }
            
        
        }
        
    
        #region Applicable_
        private XsdTypeBoolean applicable_;
        public XsdTypeBoolean Applicable_
        {
            get
            {
                if (this.applicable_ != null)
                {
                    return this.applicable_; 
                }
                else if (this.applicableIDRef_ != null)
                {
                    applicable_ = IDManager.getID(applicableIDRef_) as XsdTypeBoolean;
                    return this.applicable_; 
                }
                else
                {
                      return this.applicable_; 
                }
            }
            set
            {
                if (this.applicable_ != value)
                {
                    this.applicable_ = value;
                }
            }
        }
        #endregion
        
        public string applicableIDRef_ { get; set; }
        #region GenerationAsset_
        private CommodityDeliveryPoint generationAsset_;
        public CommodityDeliveryPoint GenerationAsset_
        {
            get
            {
                if (this.generationAsset_ != null)
                {
                    return this.generationAsset_; 
                }
                else if (this.generationAssetIDRef_ != null)
                {
                    generationAsset_ = IDManager.getID(generationAssetIDRef_) as CommodityDeliveryPoint;
                    return this.generationAsset_; 
                }
                else
                {
                      return this.generationAsset_; 
                }
            }
            set
            {
                if (this.generationAsset_ != value)
                {
                    this.generationAsset_ = value;
                }
            }
        }
        #endregion
        
        public string generationAssetIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/ElectricityDeliveryUnitFirm.cpp
// ElectricityDeliveryUnitFirm.cpp 
#include "ElectricityDeliveryUnitFirm.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ElectricityDeliveryUnitFirm::ElectricityDeliveryUnitFirm(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //applicableNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* applicableNode = xmlNode->FirstChildElement("applicable");

   if(applicableNode){applicableIsNull_ = false;}
   else{applicableIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- applicableNode , address : " << applicableNode << std::endl;
   #endif
   if(applicableNode)
   {
      if(applicableNode->Attribute("href") || applicableNode->Attribute("id"))
      {
          if(applicableNode->Attribute("id"))
          {
             applicableIDRef_ = applicableNode->Attribute("id");
             applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));
             applicable_->setID(applicableIDRef_);
             IDManager::instance().setID(applicableIDRef_,applicable_);
          }
          else if(applicableNode->Attribute("href")) { applicableIDRef_ = applicableNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));}
   }

   //generationAssetNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* generationAssetNode = xmlNode->FirstChildElement("generationAsset");

   if(generationAssetNode){generationAssetIsNull_ = false;}
   else{generationAssetIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- generationAssetNode , address : " << generationAssetNode << std::endl;
   #endif
   if(generationAssetNode)
   {
      if(generationAssetNode->Attribute("href") || generationAssetNode->Attribute("id"))
      {
          if(generationAssetNode->Attribute("id"))
          {
             generationAssetIDRef_ = generationAssetNode->Attribute("id");
             generationAsset_ = boost::shared_ptr<CommodityDeliveryPoint>(new CommodityDeliveryPoint(generationAssetNode));
             generationAsset_->setID(generationAssetIDRef_);
             IDManager::instance().setID(generationAssetIDRef_,generationAsset_);
          }
          else if(generationAssetNode->Attribute("href")) { generationAssetIDRef_ = generationAssetNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { generationAsset_ = boost::shared_ptr<CommodityDeliveryPoint>(new CommodityDeliveryPoint(generationAssetNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> ElectricityDeliveryUnitFirm::getApplicable()
{
   if(!this->applicableIsNull_){
        if(applicableIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(applicableIDRef_));
        }else{
             return this->applicable_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<CommodityDeliveryPoint> ElectricityDeliveryUnitFirm::getGenerationAsset()
{
   if(!this->generationAssetIsNull_){
        if(generationAssetIDRef_ != ""){
             return boost::shared_static_cast<CommodityDeliveryPoint>(IDManager::instance().getID(generationAssetIDRef_));
        }else{
             return this->generationAsset_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityDeliveryPoint>();
   }
}
}

